packages/caliper-core/test/common/core/transaction-statisitcs-collector.js

```diff
@@ -0,0 +1,219 @@
+            transactionStatisticsCollector.activate();
+
+            const updateSpy = sandbox.spy(transactionStatisticsCollector, '_updateStatistics');
+
+            transactionStatisticsCollector.txSubmitted(1);
+            transactionStatisticsCollector.getTotalSubmittedTx().should.equal(1);
+
+            clock.tick(50);
+            const txStatus = new TxStatus();
+            transactionStatisticsCollector.txFinished(txStatus);
+            sinon.assert.calledOnce(updateSpy);
+        });
+
+        it('should be possible to update a TransactionStatisticsCollector with an array of TxStatus results once activated', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+            transactionStatisticsCollector.activate();
+
+            const updateSpy = sandbox.spy(transactionStatisticsCollector, '_updateStatistics');
+
+            transactionStatisticsCollector.txSubmitted(1);
+            transactionStatisticsCollector.getTotalSubmittedTx().should.equal(1);
+
+            clock.tick(50);
+            const txStatus = new TxStatus();
+            transactionStatisticsCollector.txFinished([txStatus, txStatus]);
+            sinon.assert.calledTwice(updateSpy);
+        });
+
+        it('should not be possible to update a TransactionStatisticsCollector with a TxStatus result once deactivated', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+            transactionStatisticsCollector.activate();
+
+            const updateSpy = sandbox.spy(transactionStatisticsCollector, '_updateStatistics');
+
+            transactionStatisticsCollector.txSubmitted(1);
+            transactionStatisticsCollector.getTotalSubmittedTx().should.equal(1);
+
+            clock.tick(50);
+            const txStatus = new TxStatus();
+            transactionStatisticsCollector.txFinished(txStatus);
+            clock.tick(25);
+            transactionStatisticsCollector.deactivate();
+            transactionStatisticsCollector.txFinished(txStatus);
+
+            sinon.assert.calledOnce(updateSpy);
+            transactionStatisticsCollector.getRoundFinishTime().should.equal(75);
+        });
+
+    });
+
+    describe('Updating a TransactionStatisticsCollector with a TxStatus result', () => {
+
+        it('should update successful transactions', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+            transactionStatisticsCollector.activate();
+
+            clock.tick(50);
+            const txStatus = new TxStatus();
+            clock.tick(50);
+            txStatus.SetStatusSuccess();
+
+            transactionStatisticsCollector.txFinished(txStatus);
+
+            transactionStatisticsCollector.getTotalFinishedTx().should.equal(1);
+            transactionStatisticsCollector.getTotalSuccessfulTx().should.equal(1);
+            transactionStatisticsCollector.getFirstCreateTime().should.equal(50);
```

Comment:
It might be better to use the `txStatus.GetXY()` return values for the expected values, rather than literal numbers. That will clearly indicate the source of the expected value. (Only applies if a metric is derived from txStat metrics.)